ccxt_backtesting_exchange/backtester.py

```diff
@@ -30,9 +53,9 @@ def __init_balances(self, balances: Dict):
             [
                 {
                     "asset": asset,
-                    "free": balance,
-                    "used": 0,
-                    "total": balance,
+                    "free": float(balance),
```

Comment:
I think we should have a rethink on the usage of float, as there is a lot of issues around money calculation and precision generally.

There are 2 ways I know to resolve this:
1. Mulitply by the denomination then divide final value by the denomination, this way we do the calculation in the smallest unit then revert back to the actual unit, just like dividing converting dollars to cents then you perform addiontion and subtraction then revert the final value back to dollar.
2. we can also use the decimal module to avoid the overhead of the denomination conversions, 

you can check more here 
- https://learnpython.com/blog/count-money-python/
- https://www.youtube.com/watch?v=0kzjD6jvfnk

Reply:
I agree, but it's only using float here before we convert it all into numpy datatype. Let me even remove the type casting and it shoudl still work.

I'll just uncast it.

Reply:
I'm no longer forcing cast to float.

However do yu still stink it'll be a problem if we're using 2.3 without casting to Decimal, since once we pass it into pandas DataFrame it'll convert it to a numpy type.